CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+into the School of Computing (SoC).
+
+## Contributing to the document
+
+As much as possible, we want to adhere to the the overarching ideas in the
+[awesome manifesto](https://github.com/sindresorhus/awesome/blob/main/awesome.md).
+
+In other words:
+- It's curation, not collection. - We want to avoid the document from being too lengthy or diluted.
+- Research if the stuff you're including is actually *awesome*. - Only put stuff in that you or another contributor can **personally** recommend.
+- You should rather leave stuff out than include too much. - If you're unsure, ask someone else, or comment it in the PR.
+- Comment on why something is *awesome* (Not relevant for advice) - If you're recommending a book, or a lecture, tell us why you think it's great.
+- Pay attention to grammar. - Ensure your stuff is grammatically correct, typo-free and has no Markdown formatting errors.
+
+## I'm ready to contribute! How do I start?
+
+**Make sure your content follows the general principles in [Contributing to the document](#contributing-to-the-document)**
+
+### If you're giving general advice and tips to freshmen
+- [ ] Remember to follow the [Code of Conduct](CODE-OF-CONDUCT.md). We will not approve
+anything that goes against the Code of Conduct, or is submitted in bad faith.
+- [ ] Make sure there is no grammatical errors.
+- [ ] There is no specific topic or format you have to follow, but we recommend you to
+try and give advice on certain topics or areas that other contributors might
+have missed out on. (It doesn't just have to be about school life, anything from general
+programming tips, career advice, life advice that you live by and think would be helpful
+works!)
+
+### If you want to suggest a new section/list (e.g. favourite programming languages to learn, etc.)
+- [ ] Make sure that the section is relevant to the document. If you're not sure, feel free to 
+add an issue to ask us.
+- [ ] Make sure that the section is not already covered in the document.
+- [ ] Make sure there is no grammatical errors.
+- [ ] You should explain why you think the section is great.
+
+### If you're adding on to a section/list that already exists
+- [ ] Make sure that the item you're adding belongs in the relevant section/list.
+- [ ] Make sure that the item you're adding is not already covered in the document.
+- [ ] Make sure there is no grammatical errors.
+- [ ] You should briefly explain
+    - [ ] Why you think the item is great
+    - [ ] What relevance it has to the section/list
+(Feel free to leave this out if it is obvious).
+
+## How to contribute
+
+- Create a new fork of the document.
+- In your own fork, edit the document and commit it.
+    - If you're unfamiliar with markdown, try [this](https://www.markdowntutorial.com).
+- Create a Pull Request (PR) to the main document. See the Pull Request Checklist below.
+- We'll try to approve it as soon as possible!
```

Comment:
```suggestion
- We'll try to review it as soon as possible!
```
Just to be clear that it's not necessary that a PR submitted and reviewed will be approved